Sources/Swift/FlexLayout.swift

```diff
@@ -69,15 +72,23 @@ public final class Flex {
     }
     
     /**
-     This method is similar to `addItem(: UIView)` except that it also creates the flex item's UIView. Internally the method creates an
-     UIView, adds it has subviews and enables flexbox. This is useful to add a flex item/container easily when you don't need to refer to it later.
+     Adds a flex item (`UIView`) to the receiver and returns the item's flex interface.
+    
+     This method enables flexbox for `view` and adds it as a subview of the receiver's associated host view.
     
-     - Parameter view: view to add to the flex container
-     - Returns: The added view flex interface
+     Just as views can only have one superview, Yoga nodes should only have one parent node. If `view`
+     already has a superview, this method removes `view`'s associated yoga node as a child from the superview's
+     associated yoga node.
+    
+     - Parameter view: The view to be added.
+     - Returns: The flex interface corresponding to the added view.
      */
     @discardableResult
     public func addItem(_ view: UIView) -> Flex {
         if let host = self.view {
+            if let superview = view.superview, superview != host && superview.isYogaEnabled {
+                YGNodeRemoveChild(superview.yoga.node, view.yoga.node)
+            }
```

Comment:
Fix is here - check if `view` currently has a superview that isn't the host view, and if so remove it's associated yoga node as a child from the superview's associated yoga node.